onnxruntime/test/onnx/main.cc

```diff
@@ -627,12 +627,13 @@ select from 'TF8', 'TF16', 'UINT8', 'FLOAT', 'ITENSOR'. \n)");
     {"BERT_Squad", "test data bug"},
     {"constantofshape_float_ones", "test data bug", {"onnx141", "onnx150"}},
     {"constantofshape_int_zeros", "test data bug", {"onnx141", "onnx150"}},
-    {"convtranspose_autopad_same", "Implementation need to be adjusted for ONNX changes"},
+    {"convtranspose_autopad_same", "Test data has been corrected in ONNX 1.10.", {"onnx180", "onnx181", "onnx190"}},
     {"cast_STRING_to_FLOAT", "Linux CI has old ONNX python package with bad test data", {"onnx141"}},
     // Numpy float to string has unexpected rounding for some results given numpy default precision is meant to be 8.
     // "e.g. 0.296140194 -> '0.2961402' not '0.29614019'. ORT produces the latter with precision set to 8,
     // which doesn't match the expected output that was generated with numpy.
     {"cast_FLOAT_to_STRING", "Numpy float to string has unexpected rounding for some results."},
+    {"cntk_simple_seg", "Bad onnx test output caused by wrong SAME_UPPER/SAME_LOWER for ConvTranspose", {}},
```

Comment:
So this model had test data that relied on ORT's incorrect ConvTranspose ?

Reply:
> So this model had test data that relied on ORT's incorrect ConvTranspose implementation 

Correct. This model has input/output for testing. Since the computation of ConvTranspose has changed when involving SAME_UPPER/SAME_LOWER, the output will be slightly different.

> Just wondering if there are models like this running in prod somewhere that will be broken due to this change. Not sure what the best way to handle this is.

Good question... It seems "fine" to me, because the change does not involve "shape" and anyway ancient models should be still "runnable". The only difference should be quite slight because of different padding values by SAME_UPPER/SAME_LOWER. Users might complain why the output is different with newer ORT, but they can still run their old models. Moreover, we have announced this behavior change in ORT 1.12.0: https://github.com/microsoft/onnxruntime/pull/11984.

> Just wondering, should the behavior change only happen starting the opset version in ONNX that clarified the spec ?

That might make sense if ONNX did opset bump when spec was clarified, but this time ONNX didn't. IMO, this computation change is more like fixing the ambiguity and keeping it in all opset_versions looks better to me. In that case, in the future people won't be confused about why old ConvTranspose behaves inversely when involving SAME_UPPER and SAME_LOWER.

Reply:
I see. Thanks.